packages/components/src/modal/Modal.tsx

```diff
@@ -0,0 +1,120 @@
+import classNames from 'classnames';
+import React, { ReactElement, ReactNode, useCallback, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+import ThemeExport from '../ThemeExport';
+
+interface ModalProps {
+  className?: string;
+  children?: ReactNode;
+  role?: string;
+  keyboard?: boolean;
+  isOpen?: boolean;
+  centered?: boolean;
+  size?: 'sm' | 'lg' | 'xl' | undefined;
+  onOpened?: () => void;
+  onClosed?: () => void;
+  toggle?: () => void;
+  'data-testid'?: string;
+}
+
+const Modal = ({
+  className = 'theme-bg-light',
+  children,
+  role = 'role',
+  keyboard = true,
+  isOpen = false,
+  centered = false,
+  size,
+  onOpened,
+  onClosed,
+  toggle,
+  'data-testid': dataTestId,
+}: ModalProps): ReactElement => {
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent): void => {
+      switch (event.key) {
+        case 'Escape':
+          if (toggle) {
+            toggle();
+          }
+          break;
+        default:
+          break;
+      }
+    },
+    [toggle]
+  );
+
+  useEffect(
+    function addKeydownEventListener() {
+      if (isOpen && keyboard) {
+        window.addEventListener('keydown', handleKeyDown);
+        return () => window.removeEventListener('keydown', handleKeyDown);
+      }
+    },
+    [handleKeyDown, isOpen, keyboard]
+  );
+
+  useEffect(
+    function open() {
+      if (isOpen && onOpened) {
+        onOpened();
+      }
+    },
+    [onOpened, isOpen]
+  );
+
+  useEffect(
+    function open() {
+      if (!isOpen && onClosed) {
+        onClosed();
+      }
+    },
+    [onClosed, isOpen]
+  );
+
+  return ReactDOM.createPortal(
+    <CSSTransition
+      appear
+      mountOnEnter
+      unmountOnExit
+      in={isOpen}
+      classNames={{
+        enterActive: 'show',
+        enterDone: 'show',
+      }}
+      timeout={ThemeExport.transitionMs}
+    >
+      <div
+        className="modal fade"
+        onClick={toggle}
+        role="dialog"
+        style={{ zIndex: 1050, display: 'block', paddingRight: '15px' }}
+      >
+        <div className={classNames('modal-backdrop fade show')} />
```

Comment:
You noticed that your transition is a little bit faster than the native bootstrap transition. If you look at the bootstrap animation at 0.1x speed, you can see that it fades in the backdrop entirely first, then when that transition is complete, the inner modal starts fading in:
https://user-images.githubusercontent.com/4505624/174670002-b1be43a2-fd50-4d8a-a5cf-f3ccf4d788f3.mp4

It looks like reactstrap starts fading/showing all at the same time. I don't know if @dsmmcken has a particular opinion about if the backdrop should fade in first, but if it should, then need to change it up a bit... Put `modal-backdrop` as a sibling of `modal`, wrap both backdrop/modal in CSSTransition, and only put the `onEnterDone` on the wrap around modal (so it only triggers the animation at the end of the transition) - so the overall structure would be something like:
```
React.createPortal(
<>
  <CSSTransition classNames={{ onEnterActive: 'show', onEnterDone: 'show' }}>
    <div className="modal-backdrop fade"></div>
  </CSSTransition>
  <CSSTransition classNames={{ onEnterDone: 'show' }} timeout={ThemeExport.transitionMs}>
    <div className="modal fade">
    </div>
  </CSSTransition>
</>
)
```